File: app/src/main/java/com/example/multimediahub/Adapter/PDFsAdapter.java
package com.example.multimediahub.Adapter;

import android.content.Context;
import android.text.format.Formatter;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.multimediahub.PDFsViewHolder;
import com.example.multimediahub.R;

import java.io.File;
import java.util.List;

public class PDFsAdapter extends RecyclerView.Adapter<PDFsViewHolder> {
    private Context context;
    private List<File> file;
    public PDFsAdapter(Context context, List<File> file){
        this.context = context;
        this.file = file ;
    }

    @NonNull
    @Override
    public PDFsViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new PDFsViewHolder(LayoutInflater.from(context).inflate(R.layout.pdfsitem, parent, false));

    }

    @Override
    public void onBindViewHolder(@NonNull PDFsViewHolder holder, int position) {
        holder.tvName.setText(file.get(position).getName());
        holder.tvName.setSelected(true);
        holder.tvSize.setText(Formatter.formatShortFileSize(context, file.get(position).length()));

        if (file.get(position).getName().toLowerCase().endsWith(".pdf")) {
            holder.imgFile.setImageResource(R.drawable.baseline_picture_as_pdf_24);
        }
        else{
            holder.imgFile.setImageResource(R.drawable.baseline_picture_as_pdf_24);
        }
        
    }

    @Override
    public int getItemCount() {
        return file.size();
    }
}

File: app/src/main/java/com/example/multimediahub/Adapter/PDFsAdapter.kt
package com.example.multimediahub.Adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.multimediahub.R
import java.io.File

class PDFsAdapter(private val context: Context, private val pdffiles: List<String>) : RecyclerView.Adapter<PDFsAdapter.PDFsViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PDFsViewHolder {
        val view: View = LayoutInflater.from(context).inflate(R.layout.imageitem, parent, false)
        return PDFsViewHolder(view)
    }


    override fun onBindViewHolder(holder: PDFsViewHolder, position: Int) {
        val path = pdffiles[position]
        val pdfFile = File(path)
        val fileName = pdfFile.name
        holder.bind(fileName)
    }
    override fun getItemCount(): Int {
        return pdffiles.size
    }

    class PDFsViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        private val filename: TextView = itemView.findViewById(R.id.textPdfName)

        fun bind(fileName: String) {
            filename.text = fileName
        }
    }
}